pandas/core/algorithms.py

```diff
@@ -453,10 +453,8 @@ def isin(comps: AnyArrayLike, values: AnyArrayLike) -> np.ndarray:
 
     comps = _ensure_arraylike(comps)
     comps = extract_array(comps, extract_numpy=True)
-    if is_categorical_dtype(comps.dtype):
-        # TODO(extension)
-        # handle categoricals
-        return cast("Categorical", comps).isin(values)
+    if is_extension_array_dtype(comps.dtype):
+        return comps.isin(values)
 
     elif is_interval_dtype(comps.dtype):
         return cast("IntervalArray", comps).isin(values)
```

Comment:
459-460 should no longer be reachable, can be removed

Reply:
good point

Reply:
Removed.

Reply:
The `test_index_series_compat` function in `pandas.tests.arithmetic.test_interval.py` raised errors after the removal. Trying to figure out why.

Reply:
I add these two lines back, still cannot pass the test. Really curious about the CI...

Reply:
Numpy changed type inference yesterday which causes some of our tests to fail, If this is the reason why your pipelines fail this would be "fixed"